gneiss/regression/tests/test_regression.py

```diff
@@ -0,0 +1,118 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import unittest
+import subprocess
+
+import pandas as pd
+import pandas.util.testing as pdt
+from skbio.util import get_data_path
+from gneiss.regression._regression import lme_regression, ols_regression
+from gneiss.regression.tests.test_ols import TestOLS
+from gneiss.regression.tests.test_mixedlm import TestMixedLM
+from qiime2.metadata import Metadata
+
+
+class TestOLSPlugin(TestOLS):
+
+    def test_ols_regression(self):
+        m = Metadata(self.metadata)
+        res = ols_regression(self.table, self.tree, m, 'real')
+        res_coef = res.coefficients()
+        exp_coef = pd.DataFrame(
+            {'Intercept': [0, 1.00],
+             'real': [1.0, 0]},
+            index=['Y1', 'Y2'])
+
+        pdt.assert_frame_equal(res_coef, exp_coef,
+                               check_exact=False,
+                               check_less_precise=True)
+        # Double check to make sure the fit is perfect
+        self.assertAlmostEqual(res.r2, 1)
+
+        # Double check to make sure residuals are zero
+        exp_resid = pd.DataFrame([[0., 0.],
+                                  [0., 0.],
+                                  [0., 0.],
+                                  [0., 0.],
+                                  [0., 0.]],
+                                 index=['s1', 's2', 's3', 's4', 's5'],
+                                 columns=['Y1', 'Y2'])
+        pdt.assert_frame_equal(exp_resid, res.residuals())
+
+    def test_ols_cli(self):
+        # TODO: Is there a better way to test this?
```

Comment:
Yes, just load the artifacts and pass them in:

```python
from qiime2.plugins.gneiss.methods import ols_regression

in_table = qiime2.Artifact.load('path/to/foo.qza')
in_tree = qiime2.Artifact.load('path/to/foo.qza')
in_metadata = qiime2.Artifact.load('path/to/foo.qza')

results = ols_regression(table=in_table, tree=in_tree, metadata=in_metadata, etc etc)
```

Note that is a `Results` object. It is a `namedtuple` so you can fetch the individual returned artifacts out easily. If it's just a single artifact coming back, you can implicitly unpack it with:

```python
result, = ols_regression(table=in_table, tree=in_tree, metadata=in_metadata, etc etc)
```

...in which case, `result` would be an artifact of the semantic type as returned by the API method based on the plugin registration.